language/golang/mistakes.go

```diff
@@ -0,0 +1,44 @@
+package golang
+
+import (
+	"context"
+	"regexp"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/util"
+)
+
+// Mistakes builds a Go repository and returns the list of mistakes found.
+func Mistakes(logger *log.Logger, repositoryPath string) (mistakes []string, err error) {
+	output, err := util.CommandWithResult(context.Background(), logger, &util.Command{
+		Command: []string{
+			"go",
+			"build",
+		},
+
+		Directory: repositoryPath,
+	})
+	if err != nil {
+		if output != "" {
+			return extractMistakes(output), nil
+		}
+
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	return nil, nil
+}
+
+// mistakesRe defines the structure of a Go mistake.
+var mistakesRe = regexp.MustCompile(`(?m)^.*:\d+:\d+:.*$`)
+
+func extractMistakes(rawMistakes string) (mistakes []string) {
+	results := mistakesRe.FindAllStringSubmatch(rawMistakes, -1)
+
+	for _, result := range results {
+		mistakes = append(mistakes, result[0])
+	}
+
+	return mistakes
```

Comment:
Since you are just accessing the first element of the result, the following should suffice: `return mistakesRe.FindAllString(rawMistakes, -1)`